src/ion_data/mod.rs

```diff
@@ -0,0 +1,76 @@
+mod ion_eq;
+mod ion_ord;
+
+use std::cmp::Ordering;
+use std::fmt::{Display, Formatter};
+
+pub(crate) use ion_eq::IonEq;
+pub(crate) use ion_ord::IonOrd;
+
+/// A wrapper for lifting Ion compatible data into using Ion oriented comparisons (versus the Rust
+/// value semantics). This enables the default semantics to be what a Rust user expects for native
+/// values, but allows a user to opt-in to Ion's structural equivalence/order.
```

Comment:
I'd like this doc comment to be a bit crisper about the motivation. Why would one want this behavior enough to wrap one's values? I happen to know your motivation from offline discussion, but I think someone encountering this construct for the first time in the wild would need to read the code to understand it. Consider adding a couple of example use cases that show why Rust value semantics can be a poor fit for working with Ion.